src/transport/AdminPairingTable.cpp

```diff
@@ -35,37 +35,99 @@ CHIP_ERROR AdminPairingInfo::StoreIntoKVS(PersistentStorageDelegate * kvs)
     char key[KeySize()];
     ReturnErrorOnFailure(GenerateKey(mAdmin, key, sizeof(key)));
 
-    StorableAdminPairingInfo info;
-    info.mNodeId   = Encoding::LittleEndian::HostSwap64(mNodeId);
-    info.mAdmin    = Encoding::LittleEndian::HostSwap16(mAdmin);
-    info.mFabricId = Encoding::LittleEndian::HostSwap64(mFabricId);
-    info.mVendorId = Encoding::LittleEndian::HostSwap16(mVendorId);
+    StorableAdminPairingInfo * info = chip::Platform::New<StorableAdminPairingInfo>();
```

Comment:
nit: I wonder if we could use ScopedMemoryBuffer to auto-free this and not need the free and 'goto exit' pattern.

A comment of 'stored in heap because certificate sizes are large' may be helpful in understanding why this moved to heap.

Reply:
I tried that actually. ScopedMemoryBuffer could not be used as it doesn't call Constructor on the objects (it just allocates the memory). `P256SerializedKeypair` was the main reason for it.